### PR TITLE
Add UserProfile.identityExists

### PR DESCRIPTION
**Pull request by webmaster128**
To avoid usage of private APIs in https://github.com/iov-one/ponferrada/blob/v0.3.1/packages/sanes-chrome-extension/src/extension/background/model/accountManager/index.ts#L107-L111

### CODE REVIEW

File: packages/iov-keycontrol/src/userprofile.spec.ts [hunk 1]
```diff
@@ -620,6 +620,35 @@ describe("UserProfile", () => {
...
+    });
+
+    it("returns true for non-existing identity", async () => {
+      const profile = new UserProfile();
+      const wallet = profile.addWallet(Ed25519HdWallet.fromMnemonic(defaultMnemonic1));
+      await profile.createIdentity(wallet.id, defaultChain, somePath);
+      expect(await profile.identityExists(wallet.id, defaultChain, somePath)).toEqual(true);
+    });
+
+    it("throws for non-exiisting wallets", async () => {
```
Comment (willclarktech) [inline packages/iov-keycontrol/src/userprofile.spec.ts]: "non-existing"

File: packages/iov-keycontrol/src/userprofile.spec.ts [hunk 2]
```diff
@@ -620,6 +620,35 @@ describe("UserProfile", () => {
...
+  describe("identityExists", () => {
+    const somePath = [Slip10RawIndex.hardened(2), Slip10RawIndex.hardened(3), Slip10RawIndex.hardened(4)];
+
+    it("returns false for non-existing identity", async () => {
+      const profile = new UserProfile();
+      const wallet = profile.addWallet(Ed25519HdWallet.fromMnemonic(defaultMnemonic1));
+      expect(await profile.identityExists(wallet.id, defaultChain, somePath)).toEqual(false);
+    });
+
+    it("returns true for non-existing identity", async () => {
```
Comment (willclarktech) [inline packages/iov-keycontrol/src/userprofile.spec.ts]: "for existing" surely?

File: packages/iov-keycontrol/src/userprofile.ts [hunk 3]
```diff
@@ -255,6 +256,32 @@ export class UserProfile {
     return this.primaryKeyring().createIdentity(walletId, chainId, options);
   }
 
+  /**
+   * Checks if an identitiy exists in the wallet with the given ID in the primary keyring
```
Comment (willclarktech) [inline packages/iov-keycontrol/src/userprofile.ts]: "identity"

File: packages/iov-keycontrol/src/userprofile.ts [hunk 4]
```diff
@@ -255,6 +256,32 @@ export class UserProfile {
...
+   * **Example usage**
+   *
+   * This allows you to detect which accounts of an HD wallet have been created. Pseudocode
+   *
+   * ```
+   * identityExists("m/44'/234'/0'") == true
+   * identityExists("m/44'/234'/1'") == true
+   * identityExists("m/44'/234'/2'") == true
+   * identityExists("m/44'/234'/3'") == false
+   * // Shows that account indices 0-2 are created.
```
Comment (willclarktech) [inline packages/iov-keycontrol/src/userprofile.ts]: "that identities with account indices 0–2 have been created"?

File: packages/iov-keycontrol/src/userprofile.ts (hunk 3 above)
Comment (webmaster128) [reply packages/iov-keycontrol/src/userprofile.ts]: The identitiy is with us for a very long time already. Kind of a tradition to spell it that way :P

https://github.com/iov-one/iov-core/pull/1202/commits/94bd268d42fddefcb1ef4547648820d5ae0c2150

**Comment by webmaster128** [comment]
All updated

**Pull request merged**